x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2159,7 +2168,7 @@ export class RulesClient {
       }
     }
 
-    return { rules: updatedRules, errors, total };
+    return { rules: updatedRules, skipped, errors, total };
   }
 
   private async bulkEditOcc<Params extends RuleTypeParams>({
```

Comment:
`bulkEditOcc` and `bulkEdit` are quite difficult to read and understand due to their size and extensive use of closures and imperative logic (functions are not pure and they imperatively modify mutable in-memory state on the fly). Let's split them into smaller methods and rely more on immutable approach.

Reply:
Refactored both methods: split them into smaller sub-methods with descriptive names, following a more immutable approach. Let me know if you think we can further split/simplify them.

Another to-do is checking if the submethods I created can be used in other methods of the client. I seen quite a lot of repeated code (for example, ensuring auth, validating API keys, etc) which can probably be used across all main methods. Not doing it now to keep the scope limited though.